docs/USER_RECORD.md

```diff
@@ -0,0 +1,932 @@
+   only, e.g. limiting a user's memory use to 1G only makes sense on a specific
+   system that has more than 1G of memory. Each object inside the `perMachine`
+   array comes with a `matchMachineId` or `matchHostname` field which indicate
+   which systems to apply the listed settings to. Note that many fields
+   accepted in the `perMachine` section can also be set at the top level (the
+   `regular` section), where they define the fallback if no matching object in
+   `perMachine` is found.
+
+4. Various fields are located in the `binding` section (a sub-sub-object of the
+   user record; an intermediary object is insert which is keyed by the machine
+   ID of the host). Fields included in this section "bind" the object to a
+   specific system. They generally include non-portable information about paths
+   or UID assignments, that are true on a specific system, but not necessarily
+   on others, and which are managed automatically by some user record manager
+   (such as `systemd-homed`). Data in this section is considered part of the
+   user record only in the local context, and is generally not ported to other
+   systems. Due to that it is not included in the reduced user record the
+   cryptographic signature defined in the `signature` section is calculated
+   on. In `systemd-homed` this section is also removed when the user's record
+   is stored in the `~.identity` file in the home directory, so that every
+   system with access to the home directory can manage these `binding` fields
+   individually. Typically, the binding section is persisted to the local disk.
+
+5. Various fields are located in the `status` section (a sub-sub-object of the
+   user record, also with an intermediary object between that is keyed by the
+   machine ID, similar to the way the `binding` section is organized). This
+   section is augmented during runtime only, and never persisted to disk. The
+   idea is that this section contains information about current runtime
+   resource usage (for example: currently used disk space of the user), that
+   changes dynamically but is otherwise immediately associated with the user
+   record and for many purposes should be considered to be part of the user
+   record.
+
+6. The `signature` section contains one or more cryptographic signatures of a
+   reduced version of the user record. This is used to ensure that only user
+   records defined by a specific source are accepted on a system, by validating
+   the signature against the set of locally accepted signature public keys. The
+   signature is calculated from the JSON user record with all sections removed,
+   except for `regular`, `privileged`, `perMachine`. Specifically, `binding`,
+   `status`, `signature` itself and `secret` are removed first and thus not
+   covered by the signature. This section is optional, and is only used when
+   cryptographic validation of user records is required (as it is by
+   `systemd-homed.service` for example).
```

Comment:
nope, all perMachine entries are signed. 

the admin of your organization might create a user record for you, with your name, credentials and so on and will assign a normal amount of resources to it on most systems, but on some specific systems he will assign more resources, because those are the machines you preferably shall work on. So on those you get 5G of disk instead of 500M as on the others, and you'll get 4000 parallel processes max instead of 200. In the JSON record he denotes that with entries in the perMachine section, matching on the machine IDs of those machines. He then signs the record with the company privkey. The pubkey matching that privkey is installed on all company systems. Thus with that record you can then get access to all machines you need to, but on some you will get more resources than on others, and this is entirely under admin control, and you cannot escape these limits and shouldn't be able to. 

Hence, "perMachine" contains primarily resource management stuff, assigned to you by your admin in context of a specific machine in a way you cannot change it (unless you possess the privkey to change and resign it, but that means you are your own admin). 

"binding" binds a record to a local machine, i.e. contains data that only makes sense in the local context. For example, if the regular and perMachine sections do not contain a UID assignment for you, then the UID is automatically assigned on first login. And that data is then stored in "binding". This assignment is entirely local in context, and not signed. (because how could it be, the local machine likely doesn't have the privkey for that).

So yeah, perMachine and binding are similar in many cases, except one is under exclusive control of the admin with the privkey, and the other is managed by homed for you with data augmenting the record in the local context.

Note that the set of fields in the regular, the perMachine and the binding section have some overlap but are distinct. i.e the uid is one of the few that can actually appear in either of the three, but "realName" for exampe only in the regular one, and resource mgmt stuff only in the regular and perMachine sections.